### PR TITLE
$timing is undefined after soap call exception

### PR DESCRIPTION
**Pull request by insekticid**
$requestBody and $responseBody could be empty - DomDocument throws Exception

### CODE REVIEW

File: src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php [hunk 1]
```diff
@@ -102,7 +102,11 @@ public function getHash()
      */
     public function getTiming()
     {
-        return $this->timing->getDuration();
+        if ($this->timing) {
```
Comment (veewee) [inline src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php]: Maybe something like this is shorter:
```php
return $this->timing ?? 0;
```

File: src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php [hunk 2]
```diff
@@ -111,11 +115,16 @@ public function getTiming()
      */
     public function getRequestBody()
     {
-        $request = new \DOMDocument();
-        $request->preserveWhiteSpace = false;
-        $request->formatOutput = true;
-        $request->loadXML($this->requestBody);
-        return $request->saveXML();
+        if ($this->requestBody) {
```
Comment (veewee) [inline src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php]: Can you invert the if and add an early return instead?
Is this actually an error, or should the message say something like 'Empty resuest'?

File: src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php [hunk 3]
```diff
@@ -133,11 +142,16 @@ public function getRequestHeaders()
      */
     public function getResponseBody()
     {
-        $response = new \DOMDocument();
-        $response->preserveWhiteSpace = false;
-        $response->formatOutput = true;
-        $response->loadXML($this->responseBody);
-        return $response->saveXML();
+        if ($this->responseBody) {
```
Comment (veewee) [inline src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php]: Can you invert the if and add an early return instead?
Is this actually an error, or should the message say something like 'Empty response'?

**Review by veewee — CHANGES_REQUESTED** [review]
Thanks for the PR. I've added some small remarks to the code.

Can you also add some tests for the changes you made in the spec file?
https://github.com/phpro/soap-client-sf2-bridge/blob/master/spec/Phpro/SoapClient/BridgeBundle/Type/SoapCallSpec.php

This is to guarantee that the class works as expected.

File: src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php (hunk 3 above)
Comment (insekticid) [inline src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php]: maybe better could be
```
try {
$request = new \DOMDocument();
...
return $request->saveXML();
} catch (\Exception $e) {
return $->getMessage();
}
```

Comment (janvernieuwe) [reply src/Phpro/SoapClient/BridgeBundle/Type/SoapCall.php]: Definitely a good idea to try-catch the formatting!

**Comment by janvernieuwe** [comment]
Hi @insekticid,

This PR has been hanging here for a while now, hence my question if you are planning to complete the requested changes?

Thanks in advance!

**Comment by insekticid** [comment]
done

**Pull request merged**